torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -2366,63 +2351,50 @@ std::vector<PredicateDomainInfo> getPredicateContigIds(
     if (shift_expr == nullptr && gather_expr == nullptr) {
       continue;
     }
-    auto consumer_root_pos = consumer_tv->domain()->rootPosOf(consumer_root_id);
-    if ((shift_expr && shift_expr->offset(consumer_root_pos) != 0) ||
-        (gather_expr && consumer_root_pos < gather_expr->windowShape().size() &&
-         gather_expr->windowShape().at(consumer_root_pos) != 1)) {
-      excluded_ids.insert(consumer_root_id);
+    if ((shift_expr && shift_expr->offset(root_i) != 0) ||
+        (gather_expr && root_i < gather_expr->windowShape().size() &&
+         gather_expr->windowShape().at(root_i) != 1)) {
+      // TODO: The following line commented out didn't have any failures, is it
+      // needed?
+      predicate_contiguity[root_i] = false;
```

Comment:
I think this is necessary. It probably just doesn't have tests. Let me add tests.

Reply:
Fixed and added a unit test (FusionContigPredicateShift). The test should fail without the fix.